tests/overview.spec.js

```diff
@@ -137,4 +189,100 @@ describe('overview usage', () => {
     form.resetFields();
     expect(form.getFieldValue('normal')).to.be('4');
   });
+
+  it('setFieldsValue and setFieldsInitialValue for nested field works', () => {
+    form.setFieldsInitialValue({
+      foo: {
+        a: {
+          x: '1',
+          y: '2',
+        },
+        b: ['3', '4'],
+      },
+    });
+    form.setFieldsValue({
+      foo: {
+        a: {
+          x: '5',
+          y: '6',
+        },
+        b: ['7', '8'],
+      },
+    });
+    Simulate.change(form.getFieldInstance('foo.a.x'));
+    Simulate.change(form.getFieldInstance('foo.a.y'));
+    Simulate.change(form.getFieldInstance('foo.b[0]'));
+    Simulate.change(form.getFieldInstance('foo.b[1]'));
+    expect(isEqual(form.getFieldValue('foo'), {
+      a: {
+        x: '5',
+        y: '6',
+      },
+      b: ['7', '8'],
+    })).to.be(true);
+    form.setFieldsValue({ 'foo.a.x': '9' });
+    Simulate.change(form.getFieldInstance('foo.a.x'));
+    expect(isEqual(form.getFieldValue('foo'), {
+      a: {
+        x: '9',
+        y: '6',
+      },
+      b: ['7', '8'],
+    })).to.be(true);
+    form.resetFields();
+    expect(isEqual(form.getFieldValue('foo'), {
```

Comment:
@yiminghe expect(...).to.eql does not handle sparse arrays (e.g. [undefined, 123]). See screenshot below.

![2016-12-22 16_15_58-will s imac - teamviewer - free license non-commercial use only](https://cloud.githubusercontent.com/assets/4775489/21419347/07949b7e-c862-11e6-8ade-f47ee0c3f82d.png)